### PR TITLE
Add cuTENSOR test

### LINKED ISSUE #2771: Add test code for cuTENSOR

### PR DESCRIPTION
**Pull request by asi1024**
Part of chainer/chainer#8540.
Fixes #2771

### CODE REVIEW

File: tests/cupy_tests/test_cutensor.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    def test_elementwise_trinary(self):
+        desc_a = cutensor.create_tensor_descriptor(self.a, ct.OP_SQRT)
+        desc_b = cutensor.create_tensor_descriptor(self.b, ct.OP_TANH)
+        desc_c = cutensor.create_tensor_descriptor(self.c, ct.OP_COS)
+
+        d = cutensor.elementwise_trinary(
+            self.alpha, self.a, desc_a, self.mode_a,
+            self.beta, self.b, desc_b, self.mode_b,
+            self.gamma, self.c, desc_c, self.mode_c,
+            op_AB=ct.OP_ADD, op_ABC=ct.OP_MUL
```
Comment (asi1024) [inline tests/cupy_tests/test_cutensor.py]: Should we test with more operation combinations?

File: tests/cupy_tests/test_cutensor.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        )
+
+        testing.assert_allclose(
+            (self.alpha * cupy.sqrt(self.a).transpose(2, 0, 1) +
+             self.beta * cupy.tanh(self.b).transpose(1, 2, 0)) *
```
Comment (toslunar) [inline tests/cupy_tests/test_cutensor.py]: ditto

File: tests/cupy_tests/test_cutensor.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        desc_c = cutensor.create_tensor_descriptor(self.c, ct.OP_ABS)
+
+        d = cutensor.elementwise_binary(
+            self.alpha, self.a, desc_a, self.mode_a,
+            self.gamma, self.c, desc_c, self.mode_c,
+            op_AC=ct.OP_MUL
+        )
+
+        testing.assert_allclose(
+            self.alpha * (1 / (1 + cupy.exp(-self.a))).transpose(2, 0, 1) *
```
Comment (toslunar) [inline tests/cupy_tests/test_cutensor.py]: ditto

File: tests/cupy_tests/test_cutensor.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        d = cutensor.elementwise_trinary(
+            self.alpha, self.a, desc_a, self.mode_a,
+            self.beta, self.b, desc_b, self.mode_b,
+            self.gamma, self.c, desc_c, self.mode_c, out=out
+        )
+
+        assert d is out
+        testing.assert_allclose(
+            self.alpha * self.a.transpose(2, 0, 1) +
+            self.beta * self.b.transpose(1, 2, 0) +
```
[the remark this answers is not in the document]
Comment (asi1024) [reply tests/cupy_tests/test_cutensor.py]: Fixed.

File: tests/cupy_tests/test_cutensor.py (hunk 2 above)
Comment (asi1024) [reply tests/cupy_tests/test_cutensor.py]: Fixed.

File: tests/cupy_tests/test_cutensor.py (hunk 3 above)
Comment (asi1024) [reply tests/cupy_tests/test_cutensor.py]: Fixed.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
@toslunar The Jenkins test `TEST=cupy-py3-cutensor` looks to work correctly. PTAL.
```
03:15:48 tests/cupy_tests/test_cutensor.py .....s...........................      [  7%]
```

**Pull request merged**